tests/basic_event_manager.rs

```diff
@@ -35,6 +35,15 @@ impl App {
         let _ = self.event_manager.run_with_timeout(100);
     }
 
+    fn iter(&mut self) {
```

Comment:
This check doesn't really fit here. This is the example that we're also referencing in the public documentation, so I'd like to keep it as close as we can to a real use life use case.

How about we move this to a separate test instead of having it part of the `App` interface? We can add a `test_iter` and move the following asserts there after initializing the subscribers.